spidermon/contrib/scrapy/monitors.py

```diff
@@ -374,6 +380,53 @@ def test_check_if_field_coverage_rules_are_met(self):
         self.assertTrue(len(failures) == 0, msg=msg)
 
 
+@monitors.name("Jobs Comparison Monitor")
+class JobsComparisonMonitor(BaseScrapyMonitor):
+    """Check for a drop in scraped item count compared to previous jobs.
+
+    You need to set the number of previous jobs to compare, using ``SPIDERMON_JOBS_COMPARISON``.
+    The default is ``0`` which disables the monitor. We use the average of the scraped items count.
+
+    You can configure which percentage of the previous item count is the minimum acceptable, by
+    using the setting ``SPIDERMON_JOBS_COMPARISON_THRESHOLD``. We expect a float number between
+    ``0.0`` and ``1.0``. The default is ``0.8``.
+
+    You can filter which jobs to compare based on their states using the
+    ``SPIDERMON_JOBS_COMPARISON_STATES`` setting. The default value is ``("finished",)``.
+    """
+
+    @property
+    def client(self):
+        return hs
+
+    @monitors.name("Should not have a big drop in item count compared to previous jobs")
+    def test_if_item_count_from_previous_jobs_have_decreased(self):
+        # 1. get the number of jobs and check if monitor is enabled
+        number_of_jobs = self.crawler.settings.getint(SPIDERMON_JOBS_COMPARISON, 0)
+        if not number_of_jobs:
+            return
+
+        # 2. get the previous jobs meeting the defined states
+        states = self.crawler.settings.getlist(
```

Comment:
**suggestion (non-blocking)** Instead of filtering just for finished status, perhaps we could have a list of tags we can use to filter.
For example, in a project, I have some jobs that can be `finished`, but because of some different reasons, I don't want to consider them as `valid`. So I set a `SUCCESS` tag on them. Give the choice to filter by tag may be interesting to avoid us to use bad data for comparison.

Reply:
Sure. This is a WIP yet. Filtering by tags was on the to-do list. Thanks, anyway.